### ISSUE TITLE
Accessibility Compliance

### ISSUE THREAD

**Issue by kylejrp** [kognise/water.css#56]
We should pick a specific level of accessibility and make sure all of our components meet that level of accessibility.

Steps:
- Choose between AA and AAA accessibility
- Add an accessibility checker to our gulp build tools and/or pull requests
- Make sure each CSS component complies with our accessibility checks
- Make sure the demo page complies with our accessibility checks before merging
- Add accessibility to the project goals

I can take care of the build tools, but I think I'm looking for @kognise and the community's opinion on what level of accessibility we should be complying with.

**Comment by kognise** [comment]
Ok, so I'll start working on this. We should keep the following in mind as well.

> The WCAG document does not recommend that Level AAA conformance be required as a general policy for entire sites because it is not possible to satisfy all Level AAA success criteria for some content.

**Comment by kylejrp** [comment]
Awesome, thanks @kognise!

I don't think it's super unreasonable to meet AAA here since water.css is pretty small in scope. We'd just have to be cognizant of things like font sizes and colors.

Here's [Bootstrap's opinion on accessibility](https://getbootstrap.com/docs/4.1/getting-started/accessibility/) - I think it's a good philosophy:
> The overall accessibility of any project built with Bootstrap depends in large part on the author’s markup, additional styling, and scripting they’ve included. However, provided that these have been implemented correctly, it should be perfectly possible to create websites and applications with Bootstrap that fulfill WCAG 2.0 (A/AA/AAA), Section 508 and similar accessibility standards and requirements.

We can't stop people from writing inaccessable content, but we could set them up for success with default water.css styles!

**Comment by kylejrp** [comment]
There was some good discussion on Reddit on some tools that could help spot check accessibility:
https://www.reddit.com/r/webdev/comments/dl8g5v/one_click_ada_compliance_tool_does_anyone_have/

**Comment by kylejrp** [comment]
We would take any pull requests that improve the accessibility of Water.css. This could be a fix for an accessibility issue, or tooling that would help detect accessibility issues.

**Comment by eel87** [comment]
Hello, I'd love to help improve the accessibility of Water.css!  I have some experience with WAI-ARIA. I'm a new developer and this would be my first contribution...I could add accessibility to `index.html` .  Is that something you're looking for or is there another way I can help?

**Comment by southerntofu** [comment]
Thank you for building water.css it's making me love the web again! I just wanted to say from what i see here the main blocker for accessibility is low contrast.

There's this new trend throughout the design world to just make things unreadable to millions of people (including me), and it unfortunately found its way into water.css in a few areas, for example the footer.

As mentioned before, an [accessibility checker](https://wave.webaim.org/report#/https://watercss.netlify.app/) can help catch such issues! @eel87 are you still interested to make water.css more accessible? That would be amazing!

**Comment by kognise** [comment]
It looks like the only contrast issue on our new demo page (according to [WAVE](https://wave.webaim.org/report#/https://deploy-preview-194--watercss.netlify.app/)) is low contrast in citations which can be easily fixed.

I'll try to find a good way of detecting these issues in our build pipeline.

**Comment by kylejrp** [comment]
I was looking at [pa11y](https://github.com/pa11y/pa11y) the other day. There's also a build plugin for Netlify that uses pa11y to warn/error on accessibility issues.

**Comment by kognise** [comment]
Yeah, I setup pa11y - any idea what the best way to simulate different prefers-color-scheme values? I'll look into it.

**Comment by jonaskuske** [comment]
Does pa11y use puppeteer or something similar to look at pages, or does it just inspect the plain HTML?

If it actually renders the page we could re-introduce the query param the original version picker offered, so you could do `?theme=(auto|dark|light)` when visiting the docs site

**Comment by jonaskuske** [comment]
Just checked their Readme and they mention puppeteer, so the query param could work.

Also [they have an `actions` array as a beta feature](https://github.com/pa11y/pa11y#actions-array-beta), where you can specify what the simulated user should do prior to testing the site. We could use this to trigger a click on Dark Theme / Light Theme!

**Comment by kognise** [comment]
@jonaskuske I believe it uses puppeteer judging by the dependency list, but right now I'm running it on a local file like `pa11y ./dist/docs/index.html` which means no query strings.

**Comment by kognise** [comment]
There's a [thread](https://github.com/puppeteer/puppeteer/issues/4906) in the puppeteer repo about emulating prefers-color-scheme, and it's possible to give pa11y a custom browser instance so I think that's the way to go.

**Comment by jonaskuske** [comment]
Alright then, hope that works!
Otherwise the actions array should probably do the trick:

```js
pa11y('./dist/docs/index.html', {
  actions: ['click element #theme-dark']
})
pa11y('./dist/docs/index.html', {
  actions: ['click element #theme-light']
})
```

**Comment by kognise** [comment]
Fixed by #199, not sure why this didn't get automatically closed.

**Comment by jonaskuske** [comment]
@kognise because issues are only closed automatically once the respective commit/PR that closed them reaches a repository's default branch, which currently is set to master

**Comment by kylejrp** [comment]
I will mention that if there are specific accessibility issues that we haven't caught with pa11y, we would love to see a new issue logged for them! It's important to make water.css as inclusive as possible. ♥
